FakeStorageRpc: NPE on copying file

#### Environment details

1. FakeStorageRpc
2. OS type and version: macOS 11.6.5
3. Java version: 11
4. google-cloud-nio version(s): 0.123.27

#### Steps to reproduce

If you use `LocalStorageHelper` to create a storage, and create a file in it using

```java
storage.create(BlobInfo.newBuilder(sourceBucket, sourcePath).build(), content);
```

It will be stored in `FakeStorageRpc` with `null` generation. If you try to copy some other file to overwrite the original one using

```java
final var request = CopyRequest.newBuilder()
            .setSource(BlobId.of(sourceBucket, sourcePath))
            .setTarget(BlobId.of(destinationBucket, destinationPath))
            .build();

storage.copy(request)
            .getResult();
```

You'll get a `NullPointerException` in the following `FakeStorageRpc`'s code

```java
long generation = 1;
if (metadata.containsKey(destKey)) {
   generation = metadata.get(destKey).getGeneration() + 1;
}
```

since `metadata.get(destKey).getGeneration()` is `null`. Checking the generation for nullability before the increment, like it is done in `write` method, should solve the issue.

#### Stack trace
```
Caused by: java.lang.NullPointerException
	at com.google.cloud.storage.contrib.nio.testing.FakeStorageRpc.openRewrite(FakeStorageRpc.java:416)
	at com.google.cloud.storage.StorageImpl.lambda$copy$22(StorageImpl.java:511)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:103)
	at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
	at com.google.cloud.storage.Retrying.run(Retrying.java:51)
```

--- a/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/testing/FakeStorageRpc.java
+++ b/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/testing/FakeStorageRpc.java
@@ -413,7 +413,10 @@ class FakeStorageRpc extends StorageRpcTestBase {
     // if this is a new file, set generation to 1, else increment the existing generation
     long generation = 1;
     if (metadata.containsKey(destKey)) {
-      generation = metadata.get(destKey).getGeneration() + 1;
+      Long storedGeneration = metadata.get(destKey).getGeneration();
+      if (null != storedGeneration) {
+        generation = storedGeneration + 1;
+      }
     }
 
     checkGeneration(destKey, generationMatch);

--- a/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/testing/LocalStorageHelperTest.java
+++ b/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/testing/LocalStorageHelperTest.java
@@ -156,4 +156,26 @@ public class LocalStorageHelperTest {
       assertThat(ois.readObject()).isEqualTo(storageOptions);
     }
   }
+
+  @Test
+  public void testCopyOperationOverwritesExistingFile() {
+    String bucket = "bucket";
+    String original = "original";
+    String replacement = "replacement";
+    byte[] originalContent = "original content".getBytes();
+    byte[] replacementContent = "replacement content".getBytes();
+
+    localStorageService.create(BlobInfo.newBuilder(bucket, original).build(), originalContent);
+    localStorageService.create(BlobInfo.newBuilder(bucket, replacement).build(), replacementContent);
+
+    final Storage.CopyRequest request = Storage.CopyRequest.newBuilder()
+        .setSource(BlobId.of(bucket, replacement))
+        .setTarget(BlobId.of(bucket, original))
+        .build();
+
+    localStorageService.copy(request)
+        .getResult();
+
+    assertThat(localStorageService.readAllBytes(BlobId.of(bucket, original))).isEqualTo(replacementContent);
+  }
 }

--- a/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/testing/LocalStorageHelperTest.java
+++ b/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/testing/LocalStorageHelperTest.java
@@ -166,16 +166,18 @@ public class LocalStorageHelperTest {
     byte[] replacementContent = "replacement content".getBytes();
 
     localStorageService.create(BlobInfo.newBuilder(bucket, original).build(), originalContent);
-    localStorageService.create(BlobInfo.newBuilder(bucket, replacement).build(), replacementContent);
+    localStorageService.create(
+        BlobInfo.newBuilder(bucket, replacement).build(), replacementContent);
 
-    final Storage.CopyRequest request = Storage.CopyRequest.newBuilder()
-        .setSource(BlobId.of(bucket, replacement))
-        .setTarget(BlobId.of(bucket, original))
-        .build();
+    final Storage.CopyRequest request =
+        Storage.CopyRequest.newBuilder()
+            .setSource(BlobId.of(bucket, replacement))
+            .setTarget(BlobId.of(bucket, original))
+            .build();
 
-    localStorageService.copy(request)
-        .getResult();
+    localStorageService.copy(request).getResult();
 
-    assertThat(localStorageService.readAllBytes(BlobId.of(bucket, original))).isEqualTo(replacementContent);
+    assertThat(localStorageService.readAllBytes(BlobId.of(bucket, original)))
+        .isEqualTo(replacementContent);
   }
 }

